### PR TITLE
[PPS] Removed static_cast in diamond raw-to-digi unpacker

### PR DESCRIPTION
**Pull request by forthommel**
#### PR description:

This PR replaces a `static_cast` of plain TOTEM-like VFAT frames into `DiamondVFATFrame` by simple free-function accessors, following the prescription in #27860.

Also, new `pps` namespace (with nested sub-spaces) is introduced in this package to gradually replace the various `ctpps...` scopes found here and there.

#### PR validation:

Code compiles and runs successfully over a few backed up `t0streamer` streams. Produces exactly the same output as for the baseline.

#### if this PR is a backport please specify the original PR: N/A

Before submitting your pull requests, make sure you followed this checklist:
- [X] verify that the PR is really intended for the chosen branch
- [X] verify that changes follow [CMS Naming, Coding, And Style Rules](http://cms-sw.github.io/cms_coding_rules.html)
- [X] verify that the PR passes the basic test procedure suggested in the [CMSSW PR instructions](https://cms-sw.github.io/PRWorkflow.html)

### CODE REVIEW

**Comment by perrotta** [comment]
please test

File: EventFilter/CTPPSRawToDigi/interface/DiamondVFATFrame.h [hunk 1]
```diff
@@ -1,49 +1,45 @@
...
-  /// get timing infromation
-  uint32_t getLeadingEdgeTime() const {
-    uint32_t time = ((data[7] & 0x1f) << 16) + data[8];
-    time = (time & 0xFFE7FFFF) << 2 |
-           (time & 0x00180000) >> 19;  //HPTDC inperpolation bits are MSB but should be LSB.... ask HPTDC designers...
-    return time;
+/**
+ * Utilitary namespace to retrieve timing/status information from diamond VFAT frame
+**/
+namespace ppsdiamondvfat {
```
[the remark this answers is not in the document]
Comment (forthommel) [reply EventFilter/CTPPSRawToDigi/interface/DiamondVFATFrame.h]: Thanks @slava77 ! Indeed, I realised c++17 makes these nested namespaces rather elegant. We now have both the `pps::diamond` and `pps::pixel` namespaces added, along with a first `ctppsXXX` → `pps::XXX` renaming.

**Comment by slava77** [comment]
@cmsbuild please test

**Comment by slava77** [comment]
+1

for https://github.com/cms-sw/cmssw/pull/27873	fcc4123
- code changes are in line with the PR description and the follow up review; except for the part related to the issue in #27860 (which is also somewhat technical), the changes are technical/code style.
- jenkins tests pass and comparisons with the baseline show no differences.

**Comment by kpedro88** [comment]
+1

**Pull request merged**